openandroidinstaller/tooling.py

```diff
@@ -497,3 +518,24 @@ def search_device(platform: str, bin_path: Path) -> Optional[str]:
     except CalledProcessError:
         logger.error("Failed to detect a device.")
         return None
+
+@add_logging("Flash custom recovery with fastboot.")
+def fastboot_flash_recovery(
```

Comment:
Is it necessary to flash the recovery here, or could we just boot it?
I would prefer that, since if something goes wrong, the recovery of the device is still intact. This reduces the risk for users (I would say).

Reply:
I agree - that is the same reason I initially brought this feature forward

Reply:
If you don't flash the recovery, you will not be able to install OTA.
In addition, the Xiaomi recovery is very poor... And if something goes wrong, as I mentioned in `notes` metadata for lavender & ginkgo, you can flash MIUI and all partitions with a script.
For Xiaomi devices, I never saw a documentation recommanding not to flash the recovery (it's even a mandatory step)

Reply:
The OTA is a very good point. Should be fine. :+1: 
Maybe we should even consider moving more phones to flashing the recovery.

Reply:
> 

I am not sure about this one - I'd still vote for only flashing if necessary, like on Samsung or Xiaomi devices, as it is easier to revert back and as seen on the OnePlus One - the bootloader can't interact with any non stock recovery. So backwards compatibility would be another issue...